### PR TITLE
Fix actor show/hide visual gpu sim

### PR DESCRIPTION
**Pull request by arth-shukla**
tested on gpu and cpu sim, checked depth camera obs as well

## **REIMPLEMENTED**
- by default hidden objs are hidden
- if hidden actor pose is changed, save to `self.before_hide_pose` so it can be applied once actor is un-hidden

### CODE REVIEW

File: mani_skill2/envs/sapien_env.py [hunk 1]
```diff
@@ -969,9 +972,6 @@ def render_human(self):
                 self._viewer.set_camera_pose(
                     self._human_render_cameras["render_camera"].camera.global_pose
                 )
-
-        for obj in self._hidden_objects:
-            obj.show_visual()
```
Comment (StoneT2000) [inline mani_skill2/envs/sapien_env.py:974]: why is this removed? Ghost objects/visuals should show up in human renders

Comment (StoneT2000) [reply mani_skill2/envs/sapien_env.py:974]: oh is this because by default it is always shown except for rendering observations? I think we should do this. The default should be it is always hidden. Only when render_human is called is it shown.

**Review by StoneT2000 — CHANGES_REQUESTED** [review]

File: mani_skill2/envs/sapien_env.py (hunk 1 above)
Comment (StoneT2000) [reply mani_skill2/envs/sapien_env.py:974]: Well to be honest it may not matter, we don't call any of the expensive operations if the object is already hidden

Comment (StoneT2000) [reply mani_skill2/envs/sapien_env.py:974]: actually ignore this. re-read ur comments

**Comment by arth-shukla** [comment]
added code to actors so if pose is updated when hidden, it changes `self.before_hide_pose` instead

**Comment by arth-shukla** [comment]
added unit test

**Review by StoneT2000 — APPROVED** [review]
